tests/Benchmarks/System.Text.JsonLab/Comparison/JsonReader/JsonReaderComparison.cs

```diff
@@ -88,5 +88,27 @@ public void ReaderLitJson()
             var json = new LitJson.JsonReader(_reader);
             while (json.Read()) ;
         }
+
+        [Benchmark]
+        public void ReaderSpanJsonUtf16()
+        {
+            SpanJson.JsonReader<char> json = new SpanJson.JsonReader<char>(_str);
```

Comment:
Can you create the reader in GlobalSetup instead? Also, we use the EncoderTarget to enumerate the encoding (currently it only has UTF-8, but if needed, we can add the UTF-16 as well). So, having separate benchmarks like this won't work (since we wuold end up running ReaderSpanJsonUtf16 twice (once for UTF-8 and once for UTF-16) and same with the other benchmark.

Is there a way to parameterize the encoding out and set it once so we can have only one benchmark? If not, I would only add the UTf8 one for now.

Reply:
I think i'll remove the Utf16 version, it's not possible to parametrize the encoding out, as neither my API supports it nor can I add a ref struct as a field in that class.

@adamsitnik Benchmark.NET does not support having generic arguments to benchmark methods at the moment, right?
Something like:
```csharp
[ParamsSource(nameof(TypeMap))]
public Type TParam;
public Type TypeMap()
{
    switch (Target)
    {
        case EncoderTarget.InvariantUtf16:
            return typeof(char);
        case EncoderTarget.InvariantUtf8:
            return typeof(byte);
    }
}
[Benchmark]
public void MyBenchmark<TParam>() where TParam: struct
{
  // still have to solve the different arguments (_str vs. _data) then.
}
```
having some switch with cases for ``EncoderTarget`` would probably influence the benchmark too much. 
In any case this benchmark is not flattering for SpanJson, apparently having whitespace in jsons makes it 2-3 slower than without whitespace (utf8json has the same problem)

Reply:
@Tornhoof BDN supports generic types with benchmarks, but not non-generic types with generic benchmark methods

However, you can add this requirement here: https://github.com/dotnet/BenchmarkDotNet/issues/745 I can't promise to add it any time soon, but I might add it to my TODO list

Reply:
@adamsitnik I don't think that's necessary, my example above is a highly specific use case to get around benchmark design and limitations in SpanJson.